solutions/gke/configconnector/gke-cluster-autopilot/gke.yaml

```diff
@@ -0,0 +1,87 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+######
+# GKE Autopilot Cluster
+# https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-shared-vpc
+apiVersion: container.cnrm.cloud.google.com/v1beta1
+kind: ContainerCluster
+metadata:
+  name: autopilot1-gke # kpt-set: ${cluster-name}
+  namespace: project-id-tier3 # kpt-set: ${project-id}-tier3
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    cnrm.cloud.google.com/state-into-spec: absent
+spec:
+  addonsConfig:
+    configConnectorConfig:
+      enabled: false
+  clusterAutoscaling:
+    enabled: true
+    autoProvisioningDefaults:
+      serviceAccountRef:
+        name: cluster-name-sa # kpt-set: ${cluster-name}-sa
+  description: GKE Autopilot Cluster
+  databaseEncryption:
+    keyName: projects/project-id/locations/northamerica-northeast1/keyRings/cluster-name-kmskeyring/cryptoKeys/cluster-name-etcd-key # kpt-set: projects/${project-id}/locations/${location}/keyRings/${cluster-name}-kmskeyring/cryptoKeys/${cluster-name}-etcd-key
+    state: ENCRYPTED
+  enableAutopilot: true
+  # binaryAuthorization:
+  #   evaluationMode: PROJECT_SINGLETON_POLICY_ENFORCE
+  # With IntranodeVisibility, pod-to-pod traffic is sent to the VPC. Autopilot clusters must have intranode visibility enabled
+  enableIntranodeVisibility: true
+  gatewayApiConfig:
+    channel: CHANNEL_STANDARD
+  initialNodeCount: 1
+  ipAllocationPolicy:
+    clusterSecondaryRangeName: podrange
+    servicesSecondaryRangeName: servicesrange
+  location: northamerica-northeast1 # kpt-set: ${location}
+  loggingConfig:
+    enableComponents:
+      - "SYSTEM_COMPONENTS"
+      - "WORKLOADS"
+  maintenancePolicy:
+    # GMT timezone
+    dailyMaintenanceWindow:
+      startTime: 05:00
+      duration: 04:00
+  masterAuthorizedNetworksConfig:
+    cidrBlocks: # kpt-set: ${master-authorized-networks-cidr}
+      - cidrBlock: 10.1.1.5/32
+        displayName: bastion
+  monitoringConfig:
```

Comment:
Do we want prometheus enabled by default or will that be a opt-in feature? 
```
monitoringConifg:
  ...
  managedPrometheus:
      enabled: true
```

Reply:
I'm not sure this is required. The stable release version is now at 1.27.3-gke.100.

Managed collection is enabled by default for the following:

[GKE Autopilot clusters](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview) running GKE version 1.25 or greater.

Reply:
Autopilot clusters already have Cloud Logging, Cloud Monitoring, and Google Cloud Managed Service for Prometheus enabled.